### ISSUE TITLE
False positive: Cannot borrow immutable local variable as mutable

### ISSUE THREAD

**Issue by knsd** [intellij-rust/intellij-rust#1667]
```rust
let mut xs: Vec<Vec<usize>> = vec![vec![1, 2], vec![3, 4]];

for elt in &mut xs {
    elt.push(0)
//  ~~~ Error here: Cannot borrow immutable local variable `elt` as mutable
}
```

**Comment by alygin** [comment]
Looks like the same problem as in #1559

**Comment by farodin91** [comment]
@vlad20012 The main problem is type inference fails.
```rust
fn main() {
  let mut b: Vec<Vec<usize>> = vec![vec![1, 2]];
  for a in &mut b {
      a
    //^ &mut Vec<usize>
  }
}```

**Comment by vlad20012** [comment]
Okaaay, everything closes on the type inference.

**Comment by farodin91** [comment]
I will open a PR with a bit clean up code
